#  Modelos Baseline

In [1]:
import subprocess, sys
from pathlib import Path
PROJECT_ROOT = Path.cwd().parent if 'notebooks' in str(Path.cwd()) else Path.cwd()
sys.path.insert(0, str(PROJECT_ROOT))
for pkg in ['pandas', 'numpy', 'scikit-learn', 'xgboost', 'lightgbm']: 
    try: __import__(pkg)
    except: subprocess.check_call([sys.executable, '-m', 'pip', 'install', '-q', pkg])

In [2]:
import pandas as pd
import numpy as np
import sys; sys.path.insert(0, str(PROJECT_ROOT / 'src'))
from modeling import train_baseline_models
from evaluation import calculate_metrics, compare_models
import warnings; warnings.filterwarnings('ignore')

DATA_PROCESSED = PROJECT_ROOT / 'data' / 'processed'
X_train = np.load(DATA_PROCESSED / 'X_train.npy')
X_test = np.load(DATA_PROCESSED / 'X_test.npy')
y_train = np.load(DATA_PROCESSED / 'y_train.npy')
y_test = np.load(DATA_PROCESSED / 'y_test.npy')

## Entrenar y Evaluar

In [3]:
results = train_baseline_models(X_train, y_train, X_test, y_test)
comparison_df = compare_models(results, y_test)
print(comparison_df)
comparison_df.to_csv(DATA_PROCESSED / 'baseline_comparison.csv', index=False)

🤖 Entrenando modelos baseline...

  Entrenando LogisticRegression...
    ✓ Completado
  Entrenando RandomForest...
    ✓ Completado
  Entrenando XGBoost...
    ✓ Completado
  Entrenando LightGBM...
    ✓ Completado
  Entrenando KNN...
    ✓ Completado
                Model  F1_Weighted  F1_Macro  Precision    Recall  Accuracy  \
0  LogisticRegression     0.839682  0.235395   0.795480  0.889084  0.889084   
1        RandomForest     0.842185  0.242149   0.841114  0.892367  0.892367   
2             XGBoost     0.844998  0.257188   0.860185  0.891898  0.891898   
3            LightGBM     0.844523  0.258584   0.852842  0.890374  0.890374   
4                 KNN     0.842161  0.266514   0.816741  0.882401  0.882401   

    ROC_AUC  
0  0.678066  
1  0.683222  
2  0.700357  
3  0.695570  
4  0.599488  


Análisis: se Entreno el modelos base y se genero la comparación inicial; la referencia de desempeño reportada para baseline arroja F1 ≈ 0.88–0.92 según algoritmo. Se detectó baja sensibilidad en clases minoritarias en los resultados base.

Conclusiones: Las métricas base son competitivas, pero el desbalance identificado requiere técnicas de balanceo para mejorar recall en las clases menos representadas.